ui/app/adapters/kv/data.js

```diff
@@ -35,6 +35,10 @@ export default class KvDataAdapter extends ApplicationAdapter {
     return this._createOrUpdate(snapshot);
   }
 
+  updateRecord(store, type, snapshot) {
```

Comment:
I like this approach, I should probably restructure the metadata adapter to do something similar.